cpp/oneapi/dal/algo/subgraph_isomorphism/backend/cpu/matching.hpp

```diff
@@ -51,9 +51,15 @@ class matching_engine {
                     inner_alloc allocator);
     virtual ~matching_engine();
 
-    void run_and_wait(global_stack<Cpu>& gstack, std::int64_t& busy_engine_count, bool main_engine);
+    void run_and_wait(global_stack<Cpu>& gstack,
+                      std::int64_t& busy_engine_count,
+                      std::int64_t& current_match_count,
+                      std::int64_t target_match_count,
+                      bool main_engine);
     solution<Cpu> get_solution();
+    std::int64_t get_match_count() const;
```

Comment:
Do we really need it in public?

Reply:
yes, I will call this function in si function